larashed/agent-go
# Implementing an automated installation script

I have added a script to automatically install the agent on Linux-OS's.

The workflow is as follows:

1. Download the latest release from GitHub
2. Verify Checksums
3. Move the release to /usr/local/bin/larashed-agent and set permissions
4. Create a run-user named larashed
5. Generate a config file in /etc/larashed/larashed.conf
6. Generate a systemd-unit for the agent that will load the config-file
7. Issue sudo-permissions to the run-user to use docker stats and other monitoring commands

The client can be uninstalled cleanly using the provided --uninstall function of the script.

The script is build to work inside the POSIX-Shell and thus should be able to be run under most UNIXoid OS's, until now i have tested:
```
- Debian 10: 	        OK
- CentOS 8:	        OK
- CentOS 7:	        OK (but agent seems to fail with: unknown network "tcp")
- RHEL8		        OK
- Ubuntu 20.04:         OK
- Ubuntu 18.04:         OK
- Ubuntu 16.04:         OK
```

The only dependencies are:
- `systemd`
- `sudo`
- `sha256sum`
- `curl` or `wget`
